server/src/main/java/org/opensearch/identity/IdentityService.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.identity;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.OpenSearchException;
+import org.opensearch.identity.noop.NoopIdentityPlugin;
+import java.util.List;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.plugins.IdentityPlugin;
+import java.util.stream.Collectors;
+
+/**
+ * Identity and access control for OpenSearch.
+ *
+ * @opensearch.experimental
+ * */
+public class IdentityService {
+    private static final Logger logger = LogManager.getLogger(IdentityService.class);
+
+    private final Settings settings;
+    private final IdentityPlugin identityPlugin;
+
+    public IdentityService(final Settings settings, final List<IdentityPlugin> identityPlugins) {
+        this.settings = settings;
+
+        if (identityPlugins.size() == 0) {
+            identityPlugin = new NoopIdentityPlugin();
+        } else if (identityPlugins.size() == 1) {
+            identityPlugin = identityPlugins.get(0);
+        } else {
+            throw new OpenSearchException(
+                "Multiple identity plugins are not supported, found: "
```

Comment:
Since this PR introduces a default module and this file enforces at most 1 `IdentityPlugin` is installed at a time there is no way to install an IdentityPlugin with this. The opensearch node fails with this error:

```
OpenSearchException[Multiple identity plugins are not supported, found: org.opensearch.identity.shiro.ShiroIdentityPlugin,org.opensearch.security.OpenSearchSecurityPlugin]
	at org.opensearch.identity.IdentityService.<init>(IdentityService.java:38)
	at org.opensearch.node.Node.<init>(Node.java:464)
	at org.opensearch.node.Node.<init>(Node.java:377)
	at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242)
	at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242)
	at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:404)
	at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:180)
	at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:171)
	at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:104)
	at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138)
	at org.opensearch.cli.Command.main(Command.java:101)
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:137)
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:103)
For complete error details, refer to the log at /Users/cwperx/Desktop/distributions/opensearch-stephen-identity/opensearch-3.0.0-SNAPSHOT/logs/opensearch.log
```

Reply:
Hi @cwperks, great catch. This can be resolved in a couple minor changes--making the identity plugin used configurable or allowing shiro to be turned off--but it may be best to wait until @peternied is available and ask him what he thinks considering much of this was his original work.

Reply:
Since moving forward the we are not intending on using [1] security plugin build out of core, I think we shouldn't make it harder for identity plugin developers to get started by requiring them to remove/uninstall the identity-shiro plugin.  I think we can revisit how this works as we get feedback from other security implementations, until then I think we should default to use the 'no op' provider.

To support this seems like we might need to move the location from module to plugin.  @scrawfor99 I think your question [2] framed this well and we can figure out how to best move forward with a recommendation from the maintainers.
 
- [1] https://github.com/opensearch-project/security/issues/2602
- [2] https://github.com/opensearch-project/OpenSearch/pull/7246#issuecomment-1527990034